discovery_engine_core/web-api/src/models.rs

```diff
@@ -0,0 +1,100 @@
+use chrono::{NaiveDateTime, Utc};
+use derive_more::Display;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use xayn_discovery_engine_ai::{Document as AiDocument, DocumentId, Embedding};
+
+/// Unique identifier of the [`Document`].
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, Serialize, Deserialize, Display)]
+#[repr(transparent)]
+#[cfg_attr(test, derive(Default))]
+pub struct Id(Uuid);
+
+impl From<Uuid> for Id {
+    fn from(uuid: Uuid) -> Self {
+        Self(uuid)
+    }
+}
+
+impl From<Id> for Uuid {
```

Comment:
:grey_question: Why do we need to convert an id to a uuid?

Reply:
we don't, forgot to remove it

Reply:
I've reused `Id` from core crate